catalog/spatial/registry/registry-federation-admin-service-impl/src/main/java/org/codice/ddf/registry/federationadmin/service/impl/FederationAdminServiceImpl.java

```diff
@@ -13,200 +13,798 @@
+            throw new FederationAdminException("Error adding local registry entry.", e);
+        }
+
+        return registryIds;
     }
 
     @Override
-    public String addRegistryEntry(String xml) throws FederationAdminException {
+    public String addRegistryEntry(Metacard metacard, Set<String> destinations)
+            throws FederationAdminException {
+        List<String> registryIds = addRegistryEntries(Collections.singletonList(metacard),
+                destinations);
+        if (CollectionUtils.isNotEmpty(registryIds)) {
+            return registryIds.get(0);
+        }
         return null;
     }
 
     @Override
-    public String addRegistryEntry(String xml, Set<String> destinations)
+    public void updateRegistryEntry(String xml) throws FederationAdminException {
+        Metacard updateMetacard = getRegistryMetacardFromString(xml);
+        updateRegistryEntry(updateMetacard);
+    }
+
+    @Override
+    public void updateRegistryEntry(String xml, Set<String> destinations)
             throws FederationAdminException {
-        return null;
+        Metacard updateMetacard = getRegistryMetacardFromString(xml);
+        updateRegistryEntry(updateMetacard, destinations);
     }
 
     @Override
-    public String addRegistryEntry(Metacard metacard) throws FederationAdminException {
-        return null;
+    public void updateRegistryEntry(Metacard updateMetacard) throws FederationAdminException {
+        updateRegistryEntry(updateMetacard, null);
     }
 
     @Override
-    public String addRegistryEntry(Metacard metacard, Set<String> destinations)
+    public void updateRegistryEntry(Metacard updateMetacard, Set<String> destinations)
             throws FederationAdminException {
-        return null;
+        validateRegistryMetacards(Collections.singletonList(updateMetacard));
+        String registryId = updateMetacard.getAttribute(RegistryObjectMetacardType.REGISTRY_ID)
+                .getValue()
+                .toString();
+
+        List<Metacard> existingMetacards =
+                this.getRegistryMetacardsByRegistryIds(Collections.singletonList(registryId));
+
+        if (existingMetacards.size() > 1) {
+            String message = "Error updating registry entry. Multiple registry metacards found.";
+
+            List<String> metacardIds = new ArrayList<>();
+            metacardIds.addAll(existingMetacards.stream()
+                    .map(Metacard::getId)
+                    .collect(Collectors.toList()));
+            LOGGER.error("{} Matching registry metacard ids: {}", message, metacardIds);
+
+            throw new FederationAdminException(message);
+        }
+        if (existingMetacards.size() == 0) {
+            String message =
+                    "Error updating registry entry. No corresponding registry entry found.";
+
+            List<String> metacardIds = new ArrayList<>();
+            metacardIds.addAll(existingMetacards.stream()
```

Comment:
I might be missing something here, but if `existingMetacards.size() == 0` why are you then calling `existingMetacards.stream()`?